### DISCUSSION TITLE
How to detect language of a wiki?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#6728] [Q&A]
I'm wrting scripts for https://github.com/tiddly-gittly/Awesome-TiddlyWiki-Editions based on discussion in https://talk.tiddlywiki.org/t/choosing-starting-template-a-tw-edition-wiki-template-list-api-json-example/2974

But I found problem detecting language using:

```
const htmlContent = await fs.readFile(`${distEditionDir}/${distEditionName}`, 'utf8');
const tiddlers = $tw.wiki.deserializeTiddlers('text/html', htmlContent)
const result = tiddlers.filter(item => item.title?.startsWith('$:/language'))
```

![截屏2022-06-17 22 12 52](https://user-images.githubusercontent.com/3746270/174315543-82288754-51e4-45d8-a825-1c11837f1add.png)

But result is empty `[]`.

**Comment by pmario** [comment]
see: https://tiddlywiki.com/#%24%3A%2Flanguage

**Reply by linonetwo** [reply]
> see: https://tiddlywiki.com/#%24%3A%2Flanguage

see my question, `const result = tiddlers.filter(item => item.title?.startsWith('$:/language'))` returns empty `[]`.

**Reply by linonetwo** [reply]
At least this tiddler don't exist in a HTML wiki that loaded via `$tw.wiki.deserializeTiddlers`

**Comment by linonetwo** [comment]
Seems language pack plugin is not tiddler

```
{"title":"$:/languages/zh-Hans","name":"zh-Hans","plugin-type":"language","description":"Chinese (Simplified)","author":"BramChen","core-version":">=5.0.8","plugin-priority":"100","version":"5.2.3-prerelease-2022-04-09","dependents":"","type":"application/json","text":"{\"tiddlers\":{\"$:/language/Buttons/AdvancedSearch/Caption\":{\"t
```

So using `$tw.wiki.deserializeTiddlers` can't get them.

**Comment by Jermolene** [comment]
Hi @linonetwo which language tiddler are you trying to access? `$:/language` should be present in single file wikis that have had their language set via the dropdown in the "tools" sidebar. In the absence of a valid setting in `$:/language` there wouldn't be a language explicitly associated with the wiki.
